src/modules/scraper/adapter/gas-fees/gas-fees-service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from "@nestjs/common";
+import BigNumber from "bignumber.js";
+
+import { fixedPointAdjustment } from "../..//utils";
+import { EthProvidersService } from "../../../web3/services/EthProvidersService";
+import { chainIdToInfo } from "../../../../utils";
+import { MarketPriceService } from "../../../market-price/services/service";
+
+@Injectable()
+export class GasFeesService {
+  constructor(private providers: EthProvidersService, private marketPriceService: MarketPriceService) {}
+
+  public async getFillTxNetworkFee(destinationChainId: number, fillTxHash: string) {
+    const destinationChainProvider = this.providers.getProvider(destinationChainId);
+    const destinationChainInfo = chainIdToInfo[destinationChainId];
+
+    if (!destinationChainProvider || !destinationChainInfo) {
+      return {
+        fillTxBlockNumber: 0,
+        fee: "0",
+        feeUsd: "0",
+      };
+    }
+
+    const fillTxReceipt = await destinationChainProvider.getTransactionReceipt(fillTxHash);
+    // Some chains, e.g. Optimism, do not return the effective gas price in the receipt. We need to fetch it separately.
+    const gasPrice = fillTxReceipt.effectiveGasPrice || (await destinationChainProvider.getGasPrice());
+    const fillTxGasCostsWei = gasPrice.mul(fillTxReceipt.gasUsed).toString();
```

Comment:
Is there a reason for which this naming is plural? "costs"

Reply:
Changed here https://github.com/across-protocol/scraper-api/commit/5122a949c216e39ab458dc3c44e4bd81d577abc3